src/runtime/particle-check.ts

```diff
@@ -65,16 +65,25 @@ export interface ProvideSlotConnectionSpecInterface extends ConsumeSlotConnectio
 }
 
 export class Check {
-  constructor(readonly target: CheckTarget, readonly expression: CheckExpression) {}
```

Comment:
Would it make sense to extend `CheckTarget` to have the fields?

Reply:
CheckTarget is a typedef (see my reply to your comment below), so I can't extend it.

I could make it a real class?

Reply:
If it makes sense. Otherwise, fine to leave it as is.

Reply:
It's not used in very many places (mainly just this file), so I'll leave it as-is.